### ISSUE TITLE
ARM64 Support for ImmuDB

### ISSUE THREAD

**Issue by mprimeaux** [codenotary/immudb#1781]
**What would you like to be added or enhanced**

I'd very much like to see ARM64 support for ImmuDB.

**Why is this needed**

Our production node groups are all ARM64 as this represents a 40% cost per transaction savings for our workloads.

**Additional context**

For local development, we use Minikube + QEMU as the driver on our ARM64 (Apple M1 and M2 MacBook Pros) and ImmuDB doesn't deploy successfully. I receive the "classic" error when the CPU architecture of the technology doesn't support ARM64:

```shell
exec /usr/sbin/immudb: exec format error
Stream closed EOF for go-scriptures/core-db-2-immudb-0 (immudb)
```

I checked DockerHub and it seems AMD64 is the only supported architecture. What can I do to help support ARM64?

**Comment by jeroiraz** [comment]
> **What would you like to be added or enhanced**
> 
> I'd very much like to see ARM64 support for ImmuDB.
> 
> **Why is this needed**
> 
> Our production node groups are all ARM64 as this represents a 40% cost per transaction savings for our workloads.
> 
> **Additional context**
> 
> For local development, we use Minikube + QEMU as the driver on our ARM64 (Apple M1 and M2 MacBook Pros) and ImmuDB doesn't deploy successfully. I receive the "classic" error when the CPU architecture of the technology doesn't support ARM64:
> 
> ```shell
> exec /usr/sbin/immudb: exec format error
> Stream closed EOF for go-scriptures/core-db-2-immudb-0 (immudb)
> ```
> 
> I checked DockerHub and it seems AMD64 is the only supported architecture. What can I do to help support ARM64?

have you tried with this binary? https://github.com/codenotary/immudb/releases/download/v1.5.0/immudb-v1.5.0-darwin-arm64

we may need to check the docker files / release workflows in order to build docker images for that...

**Comment by mprimeaux** [comment]
How would you suggest we deploy the above-mentioned binary in Kubernetes via the existing Helm chart?

**Comment by jeroiraz** [comment]
> While the above-mentioned binary works on an ARM64 host, we’re in k8s and so would need the arm64 container images hosted in a registry. Is there any help I can provide to assist in having the arm64 container images generated as part of your usual release process?

What would be needed? A different DockerFile and a different command to build the arm64 image from it? You might be able to provide it as a basis for discussion, even in the case it's not included in the official release, the dockerFile could take the official immudb binary for arm64

@SimoneLazzaris ?

**Comment by mprimeaux** [comment]
I certainly don’t mind contributing and am also happy to provide an example GitHub Actions workflow. Should I assume [this](https://github.com/codenotary/immudb/blob/1a9282b1a481d50fa216406b7a865e4273a4a826/.github/workflows/push.yml#L222) is the step to model?

**Comment by SimoneLazzaris** [comment]
> What would be needed? A different DockerFile and a different command to build the arm64 image from it? You might be able to provide it as a basis for discussion, even in the case it's not included in the official release, the dockerFile could take the official immudb binary for arm64
> 
> @SimoneLazzaris ?

I think that a simple Dockerfile would an easy addition. @mprimeaux: the workflow you mention adds the `dev` version which is built every time we push to the repo. We'll also need to add the Dockerfile in the release flow to get official releases.

**Comment by vchaindz** [comment]
Superseded by #2069. Multi-arch Docker images for `linux/amd64` and `linux/arm64` are now built via `docker buildx` in `e212ed24` ([commit](https://github.com/codenotary/immudb/commit/e212ed24)) — the workflow change pins each Dockerfile's build stage to `BUILDPLATFORM` and defaults `BUILD_ARCH` to `${TARGETARCH}` so Go cross-compiles natively per platform.

Multi-arch manifests (`codenotary/immudb`, `codenotary/immuadmin`, `codenotary/immuclient`, plus the FIPS variants) will publish on the next master / release-tag push. Apple M1 / M2 / arm64 K8s nodes should pull arm64 images automatically once that lands — no other client change needed.

Closing as superseded; please reopen on the new ticket if anything's missing.
